test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Binders/SelectPathSegmentTokenBinderTests.cs

```diff
@@ -235,15 +230,11 @@ public override IEnumerable<IEdmOperation> FindDeclaredBoundOperations(IEdmType
             }
         }
 
-        private class FindOperationsByBindingParameterTypeHierarchyThrowingStackOverflowEdmModel : EdmModel
+        private class FindOperationsByBindingParameterTypeHierarchyThrowingNonCatchableExceptionEdmModel : EdmModel
         {
             public override IEnumerable<IEdmOperation> FindDeclaredBoundOperations(IEdmType bindingType)
             {
-#if NETCOREAPP1_0
-                throw new Exception("Oh no!");
-#else
-                throw new StackOverflowException("Oh no!");
-#endif
+                throw new OutOfMemoryException("Oh no!");

```

Comment:
Side comment: the message irked me a little because it was uninformative. :P I probably should've changed it when I saw it to give more context in some way.